lib/linguist/languages.yml

```diff
@@ -766,6 +766,122 @@ Charity:
   tm_scope: none
   ace_mode: text
   language_id: 56
+Checksums:
+  type: data
+  tm_scope: text.checksums
+  aliases:
+  - checksum
+  - hash
+  - hashes
+  - sum
+  - sums
+  filenames:
+  - CHECKSUM
+  - CHECKSUMS
+  - CKSUM
+  - CKSUMS
+  - CRC
+  - CRC32
+  - CRC32SUM
+  - CRC32SUMS
+  - CRCSUM
+  - CRCSUMS
+  - MD2
+  - MD2SUM
+  - MD2SUMS
+  - MD4
+  - MD4SUM
+  - MD4SUMS
+  - MD5
+  - MD5SUM
+  - MD5SUMS
+  - RMD160
+  - RMD160SUM
+  - RMD160SUMS
+  - SHA1
+  - SHA1SUM
+  - SHA1SUMS
+  - SHA2
+  - SHA224
+  - SHA224SUM
+  - SHA224SUMS
+  - SHA256
+  - SHA256SUM
+  - SHA256SUMS
+  - SHA2SUM
+  - SHA2SUMS
+  - SHA3
+  - SHA384
+  - SHA384SUM
+  - SHA384SUMS
+  - SHA3SUM
+  - SHA3SUMS
+  - SHA512
+  - SHA512SUM
+  - SHA512SUMS
+  - SUM
+  - SUMS
+  - checksum
+  - checksums
+  - cksum
+  - cksums
+  - crc
+  - crc32
+  - crc32sum
+  - crc32sums
+  - crcsum
+  - crcsums
+  - md2
+  - md2sum
+  - md2sums
+  - md4
+  - md4sum
+  - md4sums
+  - md5
+  - md5sum
+  - md5sums
+  - rmd160
+  - rmd160sum
+  - rmd160sums
+  - sha1
+  - sha1sum
+  - sha1sums
+  - sha2
+  - sha224
+  - sha224sum
+  - sha224sums
+  - sha256
+  - sha256sum
+  - sha256sums
+  - sha2sum
+  - sha2sums
+  - sha3
+  - sha384
+  - sha384sum
+  - sha384sums
+  - sha3sum
+  - sha3sums
+  - sha512
+  - sha512sum
+  - sha512sums
+  - sum
+  - sums
```

Comment:
Filenames, like extensions, are tested case-insensitively (although I vaguely recall there being regression-tests that *did* match filenames case-sensitively).

In any case, the entries named after a hash-function or abbreviation are generally uppercased, so let's golf this list down to:

~~~suggestion
  filenames:
  - CRC
  - CRC32
  - CRC32SUM
  - CRC32SUMS
  - CRCSUM
  - CRCSUMS
  - MD2
  - MD2SUM
  - MD2SUMS
  - MD4
  - MD4SUM
  - MD4SUMS
  - MD5
  - MD5SUM
  - MD5SUMS
  - RMD160
  - RMD160SUM
  - RMD160SUMS
  - SHA1
  - SHA1SUM
  - SHA1SUMS
  - SHA2
  - SHA224
  - SHA224SUM
  - SHA224SUMS
  - SHA256
  - SHA256SUM
  - SHA256SUMS
  - SHA2SUM
  - SHA2SUMS
  - SHA3
  - SHA384
  - SHA384SUM
  - SHA384SUMS
  - SHA3SUM
  - SHA3SUMS
  - SHA512
  - SHA512SUM
  - SHA512SUMS
  - checksum
  - checksums
  - cksum
  - cksums
  - sum
  - sums
~~~